bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/ThingActionsResource.java

```diff
@@ -217,7 +217,9 @@ public Response getActions(@PathParam("thingUID") @Parameter(description = "thin
 
     @POST
     @RolesAllowed({ Role.USER, Role.ADMIN })
-    @Path("/{thingUID}/{actionUid: [a-zA-Z0-9]+\\.[a-zA-Z0-9]+}")
+    // accept actionUid in the form of "scope.actionTypeUid" or "scope.actionTypeUid#signatureHash"
+    // # is URL encoded as %23
+    @Path("/{thingUID}/{actionUid: [a-zA-Z0-9]+\\.[a-zA-Z0-9]+(%23[a-zA-Z0-9]+)?}")
```

Comment:
The hash is a hex string, correct? In that case `A-Fa-f0-9` would be sufficient. Since we do a lookup on the hash and that is case-sensitive it Coups be a good idea to also restrict that.

Reply:
Makes sense, just implemented and pushed that.
Local test looks good.